pybamm/models/submodels/transport_efficiency/base_transport_efficiency.py

```diff
@@ -23,38 +23,26 @@ def __init__(self, param, component, options=None):
         super().__init__(param, options=options)
         self.component = component
 
-    def _get_standard_transport_efficiency_variables(
-        self, tor_n, tor_s, tor_p, set_leading_order=False
-    ):
-        tor = pybamm.concatenation(tor_n, tor_s, tor_p)
+    def _get_standard_transport_efficiency_variables(self, tor_dict):
+        component = self.component.lower()
 
-        variables = {
-            f"{self.component} transport efficiency": tor,
-            f"Positive {self.component.lower()} transport efficiency": tor_p,
-            f"X-averaged positive {self.component.lower()} "
-            "transport efficiency": pybamm.x_average(tor_p),
-        }
+        tor = pybamm.concatenation(*tor_dict.values())
 
-        if not self.half_cell:
-            variables.update(
-                {
-                    f"Negative {self.component.lower()} transport efficiency": tor_n,
-                    f"X-averaged negative {self.component.lower()} "
-                    "transport efficiency": pybamm.x_average(tor_n),
-                }
-            )
+        variables = {f"{self.component} transport efficiency": tor}
+
+        for domain, tor_k in tor_dict.items():
+            domain = domain.split()[0]
+            Domain = domain.capitalize()
```

Comment:
Can't you use the `domain_Domain` here?

Reply:
In this case we want `domain = "negative"` and `Domain = "Negative"`. `domain_Domain` would return `"negative electrode", "Negative electrode"` and then we would need to split them both.